### PR TITLE
convert library for use as a function 

### PR DESCRIPTION
**Pull request by lamchau**
This PR contains the minimal amount of changes to transform user agent strings into JSON for clickstream data. By maintaining a fork, we can update the library easier for use via snowpipe.

Modifications:
 - [x] Convert IIFE into a function
 - [x] update standalone tests (need to conditionally ignore exports)

Considerations:
 - https://github.com/lancedikson/bowser (provided less fidelity and accuracy for mobile)

Caveats:
 - Function can be pretty slow, may need consider ways to extract the regular expressions

For macOS:
`yarn build && pbcopy < dist/ua-parser.min.js`

In action: https://repl.it/repls/SpectacularMulticoloredPdf

### CODE REVIEW

File: src/ua-parser.js [hunk 1]
```diff
@@ -1113,5 +1112,13 @@
             }
         };
     }
+    return new UAParser(userAgent).getResult();
+}
 
-})(typeof window === 'object' ? window : this);
+if (typeof(exports) !== 'undefined') {
```
Comment (mhseiden) [inline src/ua-parser.js]: not sure we need this - the fn alone should be sufficient

File: src/ua-parser.js [hunk 2]
```diff
@@ -7,7 +7,7 @@
  * Dual licensed under GPLv2 or MIT
  */
 
-(function (window, undefined) {
+function parse(userAgent, window) {
```
Comment (mhseiden) [inline src/ua-parser.js]: Is `window` necessary?

**Comment by mhseiden** [comment]
LGTM modulo bits and tests.

File: src/ua-parser.js (hunk 2 above)
Comment (lamchau) [reply src/ua-parser.js]: nope, i did want to minimize the number of lines to changed when we bump versions

File: src/ua-parser.js (hunk 1 above)
Comment (lamchau) [reply src/ua-parser.js]: this is the only way i can unit test the fn and leverage the existing test cases (some). they have this gnarly [magic function](
https://github.com/faisalman/ua-parser-js/blob/732cf5834e6a8605c75f48db492a14426345d475/test/test.js#L47-L66) that taxes even more i had a hard time unpacking 

@kennethtruong do you know if it's possible to delete lines when we do a build?

Comment (mhseiden) [reply src/ua-parser.js]: Ah, no worries then!

File: src/ua-parser.js (hunk 2 above)
Comment (mhseiden) [reply src/ua-parser.js]: sgtm

**Comment by mhseiden** [comment]
LGTM!

**Pull request merged**